mesonbuild/build.py

```diff
@@ -1499,140 +1500,96 @@ def check_module_linking(self):
                 return
 
 class Generator:
-    def __init__(self, args, kwargs):
-        if len(args) != 1:
-            raise InvalidArguments('Generator requires exactly one positional argument: the executable')
-        exe = unholder(args[0])
-        if not isinstance(exe, (Executable, programs.ExternalProgram)):
-            raise InvalidArguments('First generator argument must be an executable.')
+    def __init__(self, exe: T.Union['Executable', programs.ExternalProgram],
+                 arguments: T.List[str], output: T.List[str], *,
+                 depfile: T.Optional[str] = None, capture: bool = False,
+                 depends: T.Optional[T.List[T.Union[BuildTarget, 'CustomTarget']]] = None,
+                 name: str = 'Generator'):
```

Comment:
This would be a good case for using one line for each parameter to improve readability...

Reply:
Have I ever mentioned how much I miss [attrs](https://www.attrs.org/en/stable/)?